test/wanda_web/views/v1/catalog_view_test.exs

```diff
@@ -9,16 +9,43 @@ defmodule WandaWeb.V1.CatalogViewTest do
   describe "CatalogView" do
     test "renders catalog.json" do
       checks = [
-        build(:check, expectations: build_list(3, :catalog_expectation, type: :expect)),
-        build(:check, expectations: build_list(3, :catalog_expectation, type: :expect_same))
+        build(:check, expectations: build_list(2, :catalog_expectation, type: :expect)),
+        build(:check, expectations: build_list(2, :catalog_expectation, type: :expect_same))
       ]
 
+      updated_checks =
+        Enum.map(checks, fn check ->
+          updated_expectations =
+            Enum.map(check.expectations, fn %{
+                                              name: name,
+                                              type: type,
+                                              expression: expression,
+                                              failure_message: failure_message
+                                            } ->
+              %{name: name, type: type, expression: expression, failure_message: failure_message}
+            end)
+
+          %{check | expectations: updated_expectations}
+        end)
+
+      rendered_catalog = render(CatalogView, "catalog.json", catalog: checks)
+
       assert %{
-               items: ^checks
-             } = render(CatalogView, "catalog.json", catalog: checks)
+               items: ^updated_checks
+             } = rendered_catalog
     end
   end
 
+  test "renders catalog.json like expected in the v1 schema" do
+    checks = [
+      build(:check, expectations: build_list(2, :catalog_expectation, type: :expect)),
+      build(:check, expectations: build_list(2, :catalog_expectation, type: :expect_same))
+    ]
+
+    rendered_catalog = render(CatalogView, "catalog.json", catalog: checks)
+    refute Access.get(rendered_catalog, :warning_message)
```

Comment:
Added this test as I expected more than one additional parameter that are not needed, but it was only one in the end. 
I think it does not, what do you think ?

Reply:
Some comments:
- Move this test to the `adapt to V1 version` describe and name `warning_message is removed`
- You can just use one check `build_list(1, :check)`
And the assertion could be like:
```
[%{expectations: [expectation, _]}] = render(CatalogView, "catalog.json", catalog: checks)
refute Map.has_key?(expectation, :warning_message)
```